lib/database.js

```diff
@@ -1,17 +1,41 @@
 var Collection = require('./collection')
 var bson = require('mongodb-core').BSON
 var xtend = require('xtend')
+var thunky = require('thunky')
+var toMongodbCore = require('to-mongodb-core')
+var parse = require('parse-mongo-url')
+var getTopology = require('./get-topology')
+var connect = require('./connect')
 
 var noop = function () {}
 
-var Database = function (opts, onserver) {
-  this._getServer = onserver
-  this._dbname = opts.name
-
+var Database = function (db, options) {
```

Comment:
I think it's confusing that the 1st argument is called `db` as one would think it's an instance of some higher order database-thingy, where in fact it's just an options hash (as far as I read the code at least).

But if that's the case then we suddenly have two options arguments, which is a bit confusing for the reader.

So if I understand this correctly, I think the best solution would be to merge the two and just have a single options argument. In the index.js file where you call the `Database` constructor you could then just add the `connString` and the `cols` to the `options` hash parsed in by the user, e.g:

```js
var db = new Database(extend({ connString: connString, cols: cols }, options))
```